File: pinyougouparent/pinyougousellergoodsinterface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌业务接口
 */
public interface BrandService {


    public List<TbBrand> findAll();

    public PageResult findPage(int pageNum, int pageSize);

    public PageResult findPage(TbBrand brand,int pageNum, int pageSize);

    public void add(TbBrand brand);

    public TbBrand findOne(Long id);

    public void update(TbBrand brand);

    public void delete(Long[] ids);

    public List<Map> selectOptionList();
}

File: pinyougouparent/pinyougoushopweb/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {//构建前端服务层

    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    }
    this.findPage=function(page,rows){
        return $http.get("../brand/findPage.do?page="+page+"&rows="+rows);
    }
    this.search=function (page,rows,searchEntity) {
        return $http.post("../brand/search.do?page="+page+"&rows="+rows,  searchEntity );
    }
    this.add=function(entity){
        return $http.post("../brand/add.do",entity);
    }
    this.update=function(entity){
        return $http.post("../brand/update.do",entity);
    }
    this.findOne=function (id) {
        return  $http.get("../brand/findOne.do?id="+id);
    }
    this.dele=function (ids) {
        return  $http.get("../brand/delete.do?ids="+ids);
    }
    this.selectOptionList=function(){
        return $http.get("../brand/selectOptionList.do");
    }
})